repositories/inMemory.go

```diff
@@ -0,0 +1,26 @@
+package repositories
+
+import (
+	"sync/atomic"
+	"tsm/models"
+)
+
+var (
+	DeviceRepo deviceRepoInterface = &inMemoryRepository{devices: make(map[int]models.DeviceWithId)}
+)
+
+type deviceRepoInterface interface {
```

Comment:
First of all, this interface is private - my guess is that is was supposed to be public.

Second thing is that in golang you don't have to (should not) create an interface in advance. Create the interface when you need it. This interface is needed in `services` package so I would declare it there.

In short create an interface when:
- you use something from different package and you want to be flexible about it (maybe just for tests, maybe for different implementations)
- you want to provide some flexibility and accept foreign implementation (for example in some util/generic implementation)

Third thing, to be honest I have never found an interface named like `somethingInterface` in golang.  The convention is https://golang.org/doc/effective_go.html#interfaces_and_types